### PR TITLE
Allow to reopen and/or resync metabase on SIGHUP

### LINKED ISSUE #1869: Support reloading `resync_metabase` flag with SIGHUP

### PR DESCRIPTION
**Pull request by fyrchik**
Close #1869, #1908.

### CODE REVIEW

**Review by acid-ant — APPROVED** [review]

File: pkg/local_object_storage/engine/control.go [hunk 1]
```diff
@@ -248,27 +254,31 @@ func (e *StorageEngine) Reload(rcfg ReConfiguration) error {
...
-		}
+		shardsToAdd = append(shardsToAdd, newID)
 	}
 
 	e.mtx.RUnlock()
 
 	e.removeShards(shardsToRemove...)
 
+	for _, p := range shardsToReload {
+		p.sh.Reload(p.opts...)
```
Comment (carpawell) [inline pkg/local_object_storage/engine/control.go:279]: log err here?

File: pkg/local_object_storage/metabase/control.go [hunk 2]
```diff
@@ -1,14 +1,19 @@
...
 	"path/filepath"
 
+	"github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard/mode"
 	"github.com/nspcc-dev/neofs-node/pkg/util"
 	"go.etcd.io/bbolt"
 	"go.uber.org/zap"
 )
 
+// ErrDegradedMode is returned when metabase is in a degraded mode.
+var ErrDegradedMode = errors.New("metabase is in a degraded mode")
```
Comment (carpawell) [inline pkg/local_object_storage/metabase/control.go:15]: isnt "meta" and "degraded" a confusing combination of words?

**Review by carpawell — COMMENTED** [review]
How about adding a new column to `[docs](https://github.com/nspcc-dev/neofs-node/tree/master/docs)/storage-node-configuration.md` that will indicate whether a config val could be reloaded?

File: pkg/local_object_storage/engine/control.go [hunk 3]
```diff
@@ -248,27 +254,31 @@ func (e *StorageEngine) Reload(rcfg ReConfiguration) error {
...
-			shardsToAdd = append(shardsToAdd, newID)
-		}
+		shardsToAdd = append(shardsToAdd, newID)
 	}
 
 	e.mtx.RUnlock()
 
 	e.removeShards(shardsToRemove...)
 
+	for _, p := range shardsToReload {
```
Comment (carpawell) [inline pkg/local_object_storage/engine/control.go:278]: no restrictions on shard's mode? is it safe to reload RW shard?

File: pkg/local_object_storage/engine/control.go (hunk 1 above)
Comment (fyrchik) [reply pkg/local_object_storage/engine/control.go]: Fixed

File: pkg/local_object_storage/engine/control.go (hunk 3 above)
Comment (fyrchik) [reply pkg/local_object_storage/engine/control.go:278]: Why not?

File: pkg/local_object_storage/metabase/control.go (hunk 2 above)
Comment (fyrchik) [reply pkg/local_object_storage/metabase/control.go:15]: to me they are not

File: docs/sighup.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# SIGHUP behaviour
+
+## Logger
+
+Logger level can be reloaded with a sighup.
```
[the remark this answers is not in the document]
Comment (fyrchik) [reply docs/sighup.md]: Fixed

File: pkg/local_object_storage/metabase/control.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (fyrchik) [reply pkg/local_object_storage/metabase/control.go:15]: We need it later, so added comments.

File: pkg/local_object_storage/engine/control.go (hunk 3 above)
Comment (carpawell) [reply pkg/local_object_storage/engine/control.go:278]: hm, looked at that one more time. is shard's mutex (implicitly via the `GetMode()`) the only thing that saves us from 1. refilling meta and changing the blobstor 2. closing/opening meta and performing requests to it?

File: pkg/local_object_storage/metabase/control.go (hunk 2 above)
Comment (carpawell) [reply pkg/local_object_storage/metabase/control.go:15]: ok, mb that is my problem: i thought that "degraded" was "without meta", and that is why i look at that it "meta without meta"

File: pkg/local_object_storage/engine/control.go (hunk 3 above)
Comment (fyrchik) [reply pkg/local_object_storage/engine/control.go:278]: There are no problems with _this_ line, `Reload` must be called on each shard which in turn determines whether it needs to perform any changes.

Comment (fyrchik) [reply pkg/local_object_storage/engine/control.go:278]: The problem already exists in other places, let's fix it separately https://github.com/nspcc-dev/neofs-node/issues/1956

**Review by carpawell — APPROVED** [review]

**Pull request merged**